### PR TITLE
Use `name` that return name after backticks

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/5531

### CODE REVIEW

**Comment by atulgpt** [comment]
Hi @BraisGabin here can I do something to make the patch diff pass?

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt [hunk 1]
```diff
@@ -61,6 +62,50 @@ class ConstructorParameterNamingSpec {
...
+
+    @Nested
+    inner class `with backticks` {
+        @Test
+        fun `should not complain about public param name - #5531`() {
+            val code = """
+                class Foo(val `is`: Boolean)
+            """.trimIndent()
+            val config = TestConfig(mapOf(IGNORE_OVERRIDDEN to "false"))
+            assertThat(ConstructorParameterNaming(config).compileAndLint(code)).isEmpty()
```
Comment (BraisGabin) [inline detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt:74]:
```suggestion
            assertThat(ConstructorParameterNaming().compileAndLint(code)).isEmpty()
```

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt [hunk 2]
```diff
@@ -61,6 +62,50 @@ class ConstructorParameterNamingSpec {
...
+        }
+
+        @Test
+        fun `should not complain about private param name`() {
+            val code = """
+                class Foo(private val `is`: Boolean)
+            """.trimIndent()
+            val config = TestConfig(mapOf(IGNORE_OVERRIDDEN to "false"))
+            assertThat(ConstructorParameterNaming(config).compileAndLint(code)).isEmpty()
```
Comment (BraisGabin) [inline detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt:83]:
```suggestion
            assertThat(ConstructorParameterNaming().compileAndLint(code)).isEmpty()
```

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt [hunk 3]
```diff
@@ -61,6 +62,50 @@ class ConstructorParameterNamingSpec {
...
+        }
+
+        @Test
+        fun `should complain about param name with violation`() {
+            val code = """
+                class Foo(private val `PARAM_NAME`: Boolean)
+            """.trimIndent()
+            val config = TestConfig(mapOf(IGNORE_OVERRIDDEN to "false"))
+            assertThat(ConstructorParameterNaming(config).compileAndLint(code))
```
Comment (BraisGabin) [inline detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt:92]:
```suggestion
            assertThat(ConstructorParameterNaming().compileAndLint(code))
```

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt [hunk 4]
```diff
@@ -61,6 +62,50 @@ class ConstructorParameterNamingSpec {
...
+        @Test
+        fun `should not complain about param in secondary constructor`() {
+            val code = """
+                @JvmInline
+                value class A1 constructor(val `is`: Boolean) {
+                    constructor(`is`: Boolean, `when`: Boolean): this(`is`)
+                }
+            """.trimIndent()
+            val config = TestConfig(mapOf(IGNORE_OVERRIDDEN to "false"))
+            assertThat(ConstructorParameterNaming(config).compileAndLint(code)).isEmpty()
```
Comment (BraisGabin) [inline detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt:106]:
```suggestion
            assertThat(ConstructorParameterNaming().compileAndLint(code)).isEmpty()
```

**Review by BraisGabin — COMMENTED** [review]
I think that the configuration is not important for this case, right?

**Comment by BraisGabin** [comment]
> Hi @BraisGabin here can I do something to make the patch diff pass?

No, it's ok. Your are testing this code enough.

**Comment by atulgpt** [comment]
> I think that the configuration is not important for this case, right?

Yes @BraisGabin as using `is`, `when` these as the variable name is valid code. Also we have separate rule to check unneccesary backticks

**Review by BraisGabin — APPROVED** [review]

**Comment by atulgpt** [comment]
Hi, @BraisGabin previous check got failed due to the `Stream` not allowed rule. Now I have rebased my branch. Although `codecov/patch` still failing

**Comment by BraisGabin** [comment]
Don't worry about that one.

**Comment by cortinico** [comment]
> Warnings
> ⚠️ 	This PR is approved with no milestone set. If merged, it won't appear in the Detekt release notes.
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [9362e1b](https://github.com/detekt/detekt/commit/9362e1b832b8dedb8ae419b132fc27287beb864a)

Quick heads up that those warning should not be ignored, otherwise those PRs won't make it to the changelog, while in this case this one is touching the biz logic :)

**Comment by atulgpt** [comment]
Hi @cortinico, I think I can not set the milestone. Let me know if I need to address these warnings

**Comment by cortinico** [comment]
> Hi @cortinico, I think I can not set the milestone. Let me know if I need to address these warnings

It's more a reminder for @BraisGabin and other maintainers 👍

**Pull request merged**